### PR TITLE
Improve CIS SCTP kernel module check

### PR DESCRIPTION
**Pull request by stmcginnis**
**Issue number:**

N/A

**Description of changes:**

The level 2 check 3.3.1 verifies the sctp kernel module is disabled. The current check for whether it was already loaded was not correct. There is a second check to make sure loading is also disabled. If the setting for loading is to not allow it, but the module is already laoded, the check does not return the expected output. So it would stillr eport failure, but it is less than ideal for reporting the actual issue.

This changes the check for whether the module is loaded to correctly identify if the module is loaded or not before checking whether the ability has been disabled.

**Testing done:**

With default settings:

```sh
bash-5.1# lsmod | grep sctp

bash-5.1# modprobe -n -v sctp
insmod /lib/modules/5.15.134/kernel/net/ipv4/udp_tunnel.ko.xz 
insmod /lib/modules/5.15.134/kernel/net/ipv6/ip6_udp_tunnel.ko.xz   

bash-5.1# apiclient report cis -l 2
...
[FAIL] 3.3.1     Ensure SCTP is disabled (Automatic)
...
```

With driver loaded:

```sh
bash-5.1# insmod /lib/modules/5.15.134/kernel/net/ipv4/udp_tunnel.ko.xz 
bash-5.1# insmod /lib/modules/5.15.134/kernel/net/ipv6/ip6_udp_tunnel.ko.xz 
bash-5.1# insmod /lib/modules/5.15.134/kernel/net/sctp/sctp.ko.xz 

bash-5.1# lsmod | grep sctp
sctp                  454656  6
ip6_udp_tunnel         16384  1 sctp
udp_tunnel             24576  1 sctp

bash-5.1# apiclient report cis -l 2
...
[FAIL] 3.3.1     Ensure SCTP is disabled (Automatic)
...
```

With driver loading disabled and after reboot so module could be unloaded:

```sh
bash-5.1# apiclient set kernel.modules.sctp.allowed=false
bash-5.1# reboot

bash-5.1# apiclient report cis -l 2
...
[PASS] 3.3.1     Ensure SCTP is disabled (Automatic)
...
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: sources/bloodhound/src/bin/bottlerocket-checks/checks.rs
Comment (bcressey) [inline sources/bloodhound/src/bin/bottlerocket-checks/checks.rs]: I assume your plan is to switch this to `look_for_word_in_file` once #3562 merges. The change looks good to me otherwise.

**Review by bcressey — APPROVED** [review]

File: sources/bloodhound/src/bin/bottlerocket-checks/checks.rs
Comment (stmcginnis) [reply sources/bloodhound/src/bin/bottlerocket-checks/checks.rs]: Agree - I will wait for that one to merge to add the macro, then update this one to take advantage of that.

Comment (stmcginnis) [reply sources/bloodhound/src/bin/bottlerocket-checks/checks.rs]: 3562 has merged. Rebased and updated this one to take advantage of the `look_for_word_in_file` check.

**Review by yeazelm — APPROVED** [review]

**Pull request merged**
